pkg/operators/otel-metrics/otel-metrics.go

```diff
@@ -612,6 +615,22 @@ func (m *otelMetricsOperatorInstance) PreStart(gadgetCtx operators.GadgetContext
 
 		hasValueFields := false
 
+		// Support an implicit counter
+		if implicitCounter := ds.Annotations()[AnnotationImplicitCounterName]; implicitCounter != "" {
+			tOptions := make([]metric.Int64CounterOption, 0)
+			if implicitCounterDescription := ds.Annotations()[AnnotationImplicitCounterDescription]; implicitCounterDescription != "" {
+				tOptions = append(tOptions, metric.WithDescription(implicitCounterDescription))
```

Comment:
It'd be awesome if we could provide more context in this description about when the counter is increased. For instance, if users is running this:

```bash
ig run trace_open:latest --verify-image=false --otel-metrics-name open:open --annotate open:metrics.collect=true,open:metrics.implicit-counter.name=NskubesysAndErr,open.runtime.containername:metrics.type=key -F k8s.namespace==kube-system,err!=0
```

We know that there will be a `ctrKubeSysAndErr` counter for each diff `containername` within the `kube-system` namespace and it'll increase every time a process fails to open a file. How to put that filtering information into the description? At least the `k8s.namespace==kube-system,err!=0`? It'll prevent users (like me 🤣) from naming metrics like `NskubesysAndErr` because the description will already contain that context.

I don't have that much experience with Prometheus/Grafana ... How usually people do that?
--- 

Not related with this PR: 

Issue with my idea: We have some filters with `--containername` and some others with `-F`. Another reason to merge them all into `-F`.

Reply:
> It'd be awesome if we could provide more context in this description about when the counter is increased.

IIUC, that sounds like too much magic. Even if we could do things like that (`-F` and `--containername` aren't parameters that the otel-metrics operator knows about), it might result in something that most of the users wouldn't want in the first place.
Also, IMHO if you want the metrics in the way you mentioned, I would make `err` a `metrics.type=key` as well. Then you could simply filter by that value as well in the metrics frontend (Grafana) later on (at the cost of storage, ofc).

Reply:
Fair